src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -4810,7 +4810,7 @@ public static T StdDev<T>(in ReadOnlyTensorSpan<T> x)
         {
             T mean = Average(x);
             T result = T.AdditiveIdentity;
-            TensorOperation.Invoke<TensorOperation.SumOfSquaredDifferences<T>, T, T>(x, mean, ref result);
+            TensorOperation.Invoke<TensorOperation.SumOfSquaredMagnitudeDifferences<T>, T, T>(x, mean, ref result);
```

Comment:
This approach might have an undesired performance penalty as it leads to performing the `.Abs` call for all types even when that isn't necessary to get the correct result. In `SquaredDifferenceOperator<T>` within TensorPrimitives.StdDev.cs, there is a comment indicating the Abs call is only relevant to non-vectorizable types and the `Vector128<T>`, `Vector256<T>`, and `Vector512<T>` overloads of `Invoke` do not make the call to Abs.